app/components/video-list/video-list.controller.js

```diff
@@ -0,0 +1,48 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('videoApp')
+    .controller('VideoListCtrl', class VideoListCtrl {
+      constructor(StorageService, VideoService, $mdDialog) {
+        this.StorageService = StorageService;
+        this.VideoService = VideoService;
+        this.videos = this.VideoService.videos;
+        this.options = this.VideoService.options;
+        this.$mdDialog = $mdDialog;
+      }
+
+      watchVideo(video) {
+        //console.log(video);
+        return this.$mdDialog.show({
+          templateUrl: 'app/controllers/modal-window/modal-window.template.html',
+          controller: 'ModalWindowCtrl',
+          controllerAs: 'app',
+          parent: angular.element(document.body),
+          preserveScope: true,
+          locals: {
+            video: video
+          },
+          clickOutsideToClose: true
+        });
+      }
+
+      deleteVideo(video) {
+        let videoIndex = this.videos.indexOf(video);
+        this.videos.splice(videoIndex, 1);

```

Comment:
wydaje mi się, że nie, bo StorageService tworzę na podstawie this.videos i po splice aktualizuję StorageService